pkg/vsphere/vm/vm.go

```diff
@@ -189,6 +189,25 @@ func (vm *VirtualMachine) WaitForKeyInExtraConfig(ctx context.Context, key strin
 
 	var detail string
 	var poweredOff error
+	firstFail := true
+
+	checkOptionValueFunc := func(ovs []types.BaseOptionValue) bool {
```

Comment:
Such significant number of anonymous functions make it hard to test all this by unit tests. I would consider rewriting this method to make it easier to write tests for.

Not sure it actually makes sense to change it for this PR.